### ISSUE TITLE
CLI: display the help text and exit

### ISSUE THREAD

**Issue by Zweihorn** [minetest/minetest#13121]
##### Minetest version

```
MT 5.6.1_1
```
ref the  **minetest** port from [MacPorts](https://ports.macports.org/port/minetest/)

##### OS / Hardware

Operating system: macOS 11.7.2 BigSur
CPU: x86_64

##### Summary

CLI `minetest --help` should _display the help text_ and **exit**. 

Furthermore, the help text should be amended from:
```
--help                      Show allowed options
```

supposedly to new help text:
```
--help                      Show allowed options and exit
```

and being consistent to the new _exit after help_ by the improved MT program.

However, the command **starts the full MT GUI** with a claimed MT world after the help text was displayed on Terminal i.e. macOS CLI. This should NOT happen, I presume.

Usually, commands like `tar --help` and certainly `minetest --help` should display the help text and exit without further operation. This would be similar to the `minetest --version` and more compatible with usual CLI experience.

Furthermore, might be linked to #13116 but this was _not tested_ and thus not sure about keystrokes (yet).

##### Steps to reproduce

Just give it a try on CLI like:
```
% /Applications/MacPorts/minetest.app/Contents/MacOS/minetest --help
Allowed options:
  --address <value>           Address to connect to. ('' = local game)
  --color <value>             Coloured logs ('always', 'never' or 'auto'), defaults to 'auto'
  --config <value>            Load configuration from specified file
  --console                   Starts with the console (Windows only)
  --gameid <value>            Set gameid ("--gameid list" prints available ones)
  --go                        Disable main menu
  --help                      Show allowed options
  --info                      Print more information to console
  --logfile <value>           Set logfile path ('' = no logging)
  --map-dir <value>           Same as --world (deprecated)
  --migrate <value>           Migrate from current map backend to another (Only works when using minetestserver or with --server)
  --migrate-auth <value>      Migrate from current auth backend to another (Only works when using minetestserver or with --server)
  --migrate-mod-storage <value>Migrate from current mod storage backend to another (Only works when using minetestserver or with --server)
  --migrate-players <value>   Migrate from current players backend to another (Only works when using minetestserver or with --server)
  --name <value>              Set player name
  --password <value>          Set password
  --password-file <value>     Set password from contents of file
  --port <value>              Set network port (UDP)
  --quiet                     Print to console errors only
  --random-input              Enable random user input, for testing
  --recompress                Recompress the blocks of the given map database.
  --run-benchmarks            Run the benchmarks and exit
  --run-unittests             Run the unit tests and exit
  --server                    Run dedicated server
  --speedtests                Run speed tests
  --terminal                  Feature an interactive terminal (Only works when using minetestserver or with --server)
  --trace                     Print enormous amounts of information to log and console
  --verbose                   Print even more information to console
  --version                   Show version information
  --world <value>             Set world path (implies local game if used with option --go)
  --worldlist <value>         Get list of worlds ('path' lists paths, 'name' lists names, 'both' lists both)
  --worldname <value>         Set world by name (implies local game if used with option --go)
2023-01-07 11:31:07: [Main]: Automatically selecting world at [/Users/XXX/Library/Application Support/minetest/worlds/world1]

```

However, the `minetest --version` command works as expected:
```
% /Applications/MacPorts/minetest.app/Contents/MacOS/minetest --version
Minetest 5.6.1-MacPorts-rev1 (OSX)
Using Irrlicht 1.9.0mt8
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="minetest.app/Contents/Resources"
STATIC_LOCALEDIR="minetest.app/Contents/Resources/locale"
```

Hope this helps.

Keep on the good work and prosper.

🌻

**Comment by Zweihorn** [comment]
Amend to the above was:

_(To be improved)_ supposedly to new help text:
```
--help                      Show allowed options and exit
```

and being consistent to the new _exit after help_ by the improved MT program.

**Comment by rubenwardy** [comment]
On Linux, it shows the options and exits immediately. It doesn't start the GUI

**Comment by Zweihorn** [comment]
> On Linux, it shows the options and exits immediately. It doesn't start the GUI

AFAIK the code should be the original or at least the same code as before I volunteered as new maintainer of the a.m. port. However, there are some patches coming from the legacy port which are still active. Until recently, I did not consider the effort to look into them. This may change in due time in the future acc to the above and if no other cause can be found, I presume.
